crates/sui/tests/cli_tests.rs

```diff
@@ -838,6 +838,77 @@ async fn test_package_publish_command() -> Result<(), anyhow::Error> {
     Ok(())
 }
 
+#[sim_test]
+async fn test_package_management_on_publish_command() -> Result<(), anyhow::Error> {
```

Comment:
Note: this test is largely a copy of existing `test_package_publish_command`, but different enough that it needs its own test. I'm calling out the important bits below.